src/sentry/static/sentry/app/components/charts/baseChart.tsx

```diff
@@ -227,225 +243,197 @@ type Props = {
-
-    const defaultAxesProps = {theme};
-    const yAxisOrCustom = !yAxes
-      ? yAxis !== null
-        ? YAxis({theme, ...yAxis})
-        : undefined
-      : Array.isArray(yAxes)
-      ? yAxes.map(axis => YAxis({...axis, theme}))
-      : [YAxis(defaultAxesProps), YAxis(defaultAxesProps)];
-    const xAxisOrCustom = !xAxes
-      ? xAxis !== null
-        ? XAxis({
-            ...xAxis,
-            theme,
-            useShortDate,
-            start,
-            end,
-            period,
-            isGroupedByDate,
-            utc,
-          })
-        : undefined
-      : Array.isArray(xAxes)
-      ? xAxes.map(axis =>
-          XAxis({...axis, theme, useShortDate, start, end, period, isGroupedByDate, utc})
-        )
-      : [XAxis(defaultAxesProps), XAxis(defaultAxesProps)];
-
-    // Maybe changing the series type to types/echarts Series[] would be a better solution
-    // and can't use ignore for multiline blocks
-    // @ts-expect-error
-    const seriesValid = series && series[0]?.data && series[0].data.length > 1;
-    // @ts-expect-error
-    const seriesData = seriesValid ? series[0].data : undefined;
-    const bucketSize = seriesData ? seriesData[1][0] - seriesData[0][0] : undefined;
-
-    return (
-      <ChartContainer>
-        <ReactEchartsCore
-          ref={forwardedRef}
-          echarts={echarts}
-          notMerge={notMerge}
-          lazyUpdate={lazyUpdate}
-          theme={this.props.echartsTheme}
-          onChartReady={onChartReady}
-          onEvents={this.getEventsMap}
-          opts={{
-            height,
-            width,
-            renderer,
-            devicePixelRatio,
-          }}
-          style={{
-            height: getDimensionValue(height),
-            width: getDimensionValue(width),
-            ...style,
-          }}
-          option={{
-            animation: IS_ACCEPTANCE_TEST ? false : true,
-            ...options,
-            useUTC: utc,
-            color: colors || this.getColorPalette(),
-            grid: Array.isArray(grid) ? grid.map(Grid) : Grid(grid),
-            tooltip:
-              tooltip !== null
-                ? Tooltip({
-                    showTimeInTooltip,
-                    isGroupedByDate,
-                    utc,
-                    bucketSize,
-                    ...tooltip,
-                  })
-                : undefined,
-            legend: legend ? Legend({theme, ...legend}) : undefined,
-            yAxis: yAxisOrCustom,
-            xAxis: xAxisOrCustom,
-            series: this.getSeries(),
-            axisPointer,
-            dataZoom,
-            toolbox: toolBox,
-            graphic,
-          }}
-        />
-      </ChartContainer>
-    );
-  }
+  // XXX(epurkhiser): Echarts can become unhappy if one of these event handlers
+  // causes the chart to re-render and be passed a whole different instance of
+  // event handlers.
+  //
+  // We use React.useMemo to keep the value across renders
```

Comment:
fwiw if you consider how the class version of this component worked, it would create this map upon construction and would never update. So you may consider this a marginal improvement since now the function handlers _will_ update when changed.

Reply:
Would the same error occur if for some reason these handler props were updated from the parent? I don't think this is an actual use case, but it might be good to mention this if it does error.

Reply:
I believe only if the re-render happens due to one of these handlers (though I may be wrong).

Worst case we can remove all of the dependencies from the memo list